app/(app)/vehicles/edit-vehicle.tsx

```diff
@@ -0,0 +1,86 @@
+import { Link, router, useLocalSearchParams } from 'expo-router'
+import { StatusBar } from 'expo-status-bar'
+import { useState } from 'react'
+
+import TextInput from '@/components/inputs/TextInput'
+import ScreenContent from '@/components/screen-layout/ScreenContent'
+import ScreenView from '@/components/screen-layout/ScreenView'
+import AccessibilityField from '@/components/shared/AccessibilityField'
+import Button from '@/components/shared/Button'
+import DismissKeyboard from '@/components/shared/DismissKeyboard'
+import Typography from '@/components/shared/Typography'
+import { useTranslation } from '@/hooks/useTranslation'
+import { useVehiclesStoreContext } from '@/state/VehiclesStoreProvider/useVehiclesStoreContext'
+
+export type EditVehicleParams = {
+  vehicleId: string
+}
+
+const AddVehicleScreen = () => {
+  const t = useTranslation('VehiclesScreen')
+  const { vehicleId } = useLocalSearchParams<EditVehicleParams>()
+
+  const { editVehicle, isLoading, getVehicle } = useVehiclesStoreContext()
+
+  const vehicle = getVehicle(vehicleId ? +vehicleId : null)
+
+  const [vehicleName, setVehicleName] = useState(vehicle?.name ?? '')
+
+  // If the page was reloaded or navigated to directly, then the modal should be presented as
+  // a full screen page. You may need to change the UI to account for this.
+  const isPresented = router.canGoBack()
+
+  if (!vehicleId) return null
+
+  const handleSaveVehicle = async () => {
+    if (!vehicle) return
+
+    await editVehicle({
+      id: +vehicleId,
+      licencePlate: vehicle.vehiclePlateNumber,
+      vehicleName,
+      isDefault: vehicle.isDefault,
```

Comment:
The PATCH request requires only `id` and changed attributes. This is the main difference between PATCH and PUT. Why we force to pass all attributes to this function if it is not required by the endpoint?